app/models/concerns/article.rb

```diff
@@ -31,6 +35,15 @@ def set_user
     self.user = User.current unless User.current.nil?
   end
 
+  def set_channel
+    return if self.channel.present?
+    if self.user.is_a?(BotUser) || User.current && User.current.is_a?(BotUser)
```

Comment:
Should be `self.user.is_a?(BotUser) || (User.current && User.current.is_a?(BotUser))`
as we should `||` with `User.current && User.current is bot`